developer_manual/app/js.rst

```diff
@@ -46,3 +46,37 @@ Full URLs can be generated by using:
 .. code-block:: js
 
     var authorUrl = OC.generateUrl('/apps/myapp/authors/1');
+
+
+Extending core parts
+====================
+
+It is possible to extend components of the core web UI. The following examples
+should show how this is possible.
+
+Extending the "new" menu in the files app
+-----------------------------------------
+
+.. versionadded:: 9.0
+
+.. code-block:: js
+
+    var myFileMenuPlugin = {
+        attach: function (menu) {
+            menu.addMenuEntry({
+                id: 'abc',
+                displayName: 'Menu display name',
+                templateName: 'templateName.ext',
+                iconClass: 'icon-filetype-text',
+                fileType: 'file',
+                actionHandler: function () {
+                    console.log('do something here');
+                }
+            });
+        }
+    };
+    OC.Plugins.register('OCA.Files.NewFileMenu', myFileMenuPlugin);
+
+This will register a new menu entry in the "New" menu of the files app. The
+method ``attach()`` is called once the menu is built. This usually happens right
```

Comment:
The menu is created when first clicking the "+" button. Then it is reused for subsequent clicks.
This means that the plugin is only called once.